nikola/utils.py

```diff
@@ -2093,3 +2094,171 @@ def rss_writer(rss_obj, output_path):
+        else:
+            result = []
+            for other_lang, classifications in cldata.items():
+                result.extend([(other_lang, other_classification) for other_classification in classifications])
+            return result
+
+    def has_translations(self, classification, lang):
+        """Return whether we know about the classification in that language.
+
+        Note that this function returning ``True`` does not mean that
+        ``get_translations`` returns a non-empty list, but only that
+        this classification was explicitly added with ``add_translation``
+        at some point.
+        """
+        return self._data[lang].get(classification) is not None
+
+    def add_defaults(self, posts_per_classification_per_language):
+        """Treat every classification as its own literal translation into every other language.
+
+        ``posts_per_classification_per_language`` should be the first argument
+        to ``Taxonomy.postprocess_posts_per_classification``.
+        """
+        # First collect all classifications from all languages
+        all_classifications = set()
+        for _, classifications in posts_per_classification_per_language.items():
+            all_classifications.update(classifications.keys())
+        # Next, add translation records for all of them
+        for classification in all_classifications:
+            record = {tlang: classification for tlang in posts_per_classification_per_language}
+            self.add_translation(record)
+
+    def read_from_config(self, site, basename, posts_per_classification_per_language, add_defaults_default):
+        """Read translations from config.
+
+        ``site`` should be the Nikola site object. Will consider
+        the variables ``<basename>_TRANSLATIONS`` and
+        ``<basename>_TRANSLATIONS_ADD_DEFAULTS``.
+
+        ``posts_per_classification_per_language`` should be the first argument
+        to ``Taxonomy.postprocess_posts_per_classification``, i.e. this function
+        should be called from that function. ``add_defaults_default`` specifies
+        what the default value for ``<basename>_TRANSLATIONS_ADD_DEFAULTS`` is.
+
+        Also sends signal via blinker to allow interested plugins to add
+        translations by themselves. The signal name used is
+        ``<lower(basename)>_translations_config``, and the argument is a dict
+        with entries ``translation_manager``, ``site`` and
+        ``posts_per_classification_per_language``.
+        """
+        # Add translations
+        for record in site.config.get('{}_TRANSLATIONS'.format(basename), []):
+            self.add_translation(record)
+        # Add default translations
+        if site.config.get('{}_TRANSLATIONS_ADD_DEFAULTS'.format(basename), add_defaults_default):
+            self.add_defaults(posts_per_classification_per_language)
+        # Use blinker to inform interested parties (plugins) that they can add
+        # translations themselves
```

Comment:
If someone wants a more complicated scheme (like using Google Translator to automatically translate tag/category/section/... names), they can use this hook to implement that.